src/main/java/com/hubspot/jinjava/interpret/Context.java

```diff
@@ -290,6 +291,21 @@ public void handleDeferredNode(Node node) {
     return ImmutableSet.copyOf(deferredNodes);
   }
 
+  public void handlePreservedRawTag() {
+    hasPreservedRawTags = true;
+    if (getParent() != null) {
```

Comment:
Not too sure what you mean. I am doing a recursive call, which I believe checks up the stack, excluding global: `getParent().handlePreservedRawTag();`

Reply:
Thanks, I was looking for a `while` and I missed the recursion.

Reply:
I guess `if (parent.getParent() != null) {` will execute the `handlePreservedRawTag()` for Global as well. Can you verify whether we need to have `parent.getParent().getParent() != null` to exclude the global one?

Reply:
Like in https://github.com/HubSpot/jinjava/pull/449, we don't want the shared global context to be altered here as it can be shared across threads. My goal with marking a context as `hasPreservedTags` is so that we know that a second pass must always be done if raw tags get preserved.
In the instance that we preserve all the raw tags, but nothing gets deferred, a second pass can be done when pre-rendering. If something is deferred, that second pass will happen at serve-time. I want to exclude it for global because I just want it to mark for one run of `renderForResult` as mentioned here: https://github.com/HubSpot/jinjava/pull/447.
Does that answer what you were wondering?

Reply:
Sorry, I misread that question. We only need to check if the parent of what we are calling `handlePreservedTag()` on is not null. I think maybe line 300: `getParent().handlePreservedRawTag()` is confusing, as it would make more sense for it to be `parent.handlePreservedRawTag()`.
The `if` statement is checking to make sure that `parent`'s parent is not null, if so then handle the preserved raw tag for `parent`

With a context tree shape like `G-a-b`, here's some sudo-code that uses _questionable_ syntax:
```
this == b
parent == getParent() == a
parent.getParent() == G != null
parent.handlePreservedRawTag():
-this == a
-parent == getParent() == G
-parent.getParent() == null
-return
return
```